### ISSUE TITLE
Are there any plan for using angular7?

### ISSUE THREAD

**Issue by ElderJames** [mixcore/mix.core#44]

**Comment by nhathoang989** [comment]
Hi ElderJames, 
At first, we planned to build a SSR CMS. But when we started, there was no stable solution. 
And now, after cleanup the source and release version 1, we'll back to this plan.
Btw, we're confusing between Blazor or Angular 7. Any suggestion for this ?
Thanks.

**Comment by ElderJames** [comment]
Blazor is good! Of course, I'm glad you're thinking about using it.
But so far, it doesn't have a good component library and any SSR solution.
If you have the idea of developing a good component library for Blazor,I want to get involved.

**Comment by Smilefounder** [comment]
Hi @ElderJames thanks a lot for your support. Actually, we have started the plan for Blazor 1 year ago. 

FYI: https://github.com/Swastika-IO/Swastika-IO-Admin-Blazor

However, at the time being, Blazor still having some issues and conflict with MVC, so we temporary paused the plan until it gets better. For now, it is still under "Experimental" label. But with your support, I think we can start it again soon.

**Comment by vector-man** [comment]
If you do add Angular, please don't make it mandatory; I'm not much of a fan of it. I really like Vue and React, though https://hackr.io/blog/top-angular-alternatives-2019

**Comment by Smilefounder** [comment]
Hi team, we are planning to use VueJS / Blazor in the near future :)
Angular versions keep upgrade and not sure if it is stable to start with or not. Will also considering.

Cheers!

**Comment by NickMaev** [comment]
> Hi team, we are planning to use VueJS / Blazor in the near future :)
> Angular versions keep upgrade and not sure if it is stable to start with or not. Will also considering.
> 
> Cheers!

Hi there. 

Did you do some screening of the VueJS/Blazor and their rate?
ReactJS is more popular than VueJS. And Blazor is not production ready.
What are you relying on when you plan to use something?

**Comment by Smilefounder** [comment]
Hi @NickMaev , FYI

At the moment, we are developing a totally new [Portal v2](https://github.com/mixcore/mix.portal.angular-io) with Angular 12.

I, personally think that Svelte also really potential, so I'm doing a bit R&D here: https://github.com/mixcore/mix.portal.svelte

And because all of the front-end UI frameworks going to changes and run really fast, so our team decided to develop a universal front-end library to be the core of all frameworks here https://github.com/mixcore/mix.lib.ts (this is built with Typescript). With this library. We don't need to care much about Angular / React / VueJS / Svelte / Blazor... as it will only matter at the front-end skeleton. All of the business logic of Portal will be handle at mix.lib.ts so it will easier for contributors to use any FE stacks that they would love to use.

All of these new things will focus on Mixcore v2 that we plan to release by the end of this year.

> What are you relying on when you plan to use something?

Honestly, FE frameworks now are too much and growing really fast. All of them have both pros & cons. So it really hard to decide.

However, Angular will be our first in mind at the moment because I think if we going to deal with super super enterprise platform, we should use Angular or React. But React is not in our interest (because we can not focus on both 2 big UI frameworks at the same time with .NET Core). But, I plan to develop a basic portal will all those UI frameworks for V2 + mix.lib.ts. To prove that, our mib.lib.ts is able to cover all kinds.

VueJS is cool and nice! But still not for enterprise platforms. I live Vue also because it supports both JS and TS.

Svelte is on my list since this year. However, it still really really young! Like Angular 2 at the time we decided to switch from Angular 2 to Angular JS. And wait until now to use Angular 12 instead because it is more stable and a lot of big companies are using it now.

Svelte Kit just still in Beta, and there are many issues so it still R&D for my hobby at the moment.

Blazor has been removed from our team list. As the performance still not impressed us yet. And not ready for Production (I think).

Btw, my previous comment is from Feb 2020 lol. So a lot has been changed.

Also, I would love to hear your advice as well.

Cheers!
